src/builder.rs

```diff
@@ -55,7 +55,7 @@ fn clean_history_control_dir(socketdir: &Path, prefix: &str) -> io::Result<()> {
 }
 
 /// Build a [`Session`] with options.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq)]
```

Comment:
I would not like `PartialEq` to be added to `SessionBuilder`, it should be an opaque struct and we do not provide any guarantee on how it is stored or the order of them.